packages/ketcher-core/src/domain/entities/PolymerBond.ts

```diff
@@ -104,6 +105,10 @@ export class PolymerBond extends DrawingEntity {
           secondMonomerAttachmentPoint,
         )
       ) ||
+        (isMonomerConnectedToR2RnaBase(this.firstMonomer) &&
+          this.secondMonomer instanceof RNABase) ||
+        (isMonomerConnectedToR2RnaBase(this.secondMonomer) &&
+          this.firstMonomer instanceof RNABase) ||
         firstMonomerAttachmentPoint === secondMonomerAttachmentPoint) &&
       !(
         (firstMonomerAttachmentPoint === AttachmentPointName.R1 &&
```

Comment:
Maybe at some point we could refactor this return expression by putting these logical expressions into named variables as now it is barely readable and not quite clear what happens here